src/main/java/com/nike/cerberus/cli/EnvironmentConfigToArgsMapper.java

```diff
@@ -101,147 +106,178 @@ private EnvironmentConfigToArgsMapper() {
                 return getCreateRoute53CommandArgs(environmentConfig);
             case CreateWafCommand.COMMAND_NAME:
                 return getCreateWafCommandArgs(environmentConfig);
+            case GenerateCertsCommand.COMMAND_NAME:
+                return getGenerateCertificatesCommandArgs(environmentConfig);
+            case CreateCmsCmkCommand.COMMAND_NAME:
+                return getCreateCmsCmkCommandArgs(environmentConfig);
+            case CreateEdgeDomainRecordCommand.COMMAND_NAME:
+                return getCreateEdgeDomainRecordCommandArgs(environmentConfig);
             default:
                 return new LinkedList<>();
         }
     }
 
-    private static List<String> getCreateCmsConfigCommandArgs(EnvironmentConfig environmentConfig) {
-        List<String> args = new LinkedList<>();
+    private static List<String> getCreateEdgeDomainRecordCommandArgs(EnvironmentConfig environmentConfig) {
+        return ArgsBuilder.create()
+                .addOption(CreateEdgeDomainRecordCommand.BASE_DOMAIN_NAME_LONG_ARG, environmentConfig.getBaseDomainName())
+                .addOption(CreateEdgeDomainRecordCommand.HOSTED_ZONE_ID_LONG_ARG, environmentConfig.getHostedZoneId())
+                .addOption(CreateEdgeDomainRecordCommand.EDGE_DOMAIN_NAME_OVERRIDE, environmentConfig.getEdgeDomainNameOverride())
+                .build();
+    }
 
-        ManagementService managementService = environmentConfig.getManagementService();
+    private static List<String> getCreateCmsCmkCommandArgs(EnvironmentConfig environmentConfig) {
+        ArgsBuilder args = ArgsBuilder.create();
+
+        if (environmentConfig.getManagementService().getAdditionalEncryptionCmkRegions() != null
+                && environmentConfig.getManagementService().getAdditionalEncryptionCmkRegions().size() >= 1) {
+            environmentConfig.getManagementService().getAdditionalEncryptionCmkRegions().forEach(region -> {
+                args.addOption(CreateCmsCmkCommand.ADDITIONAL_REGIONS_ARG, region);
+            });
+        } else {
+            throw new RuntimeException(String.format("%s requires at least 1 additional region be specified for high " +
```

Comment:
Isn't this validation somewhere else already?

Reply:
It's done way later in the lifecycle, I can remove this here and the composite command will fail when it get to that command rather that before the command is ran.

Reply:
To elaborate, I had to tweak isRunnable for the composite commands to not run isRunnable on every command before it tries to run then all.

previously it would do the following

c1.isRunnable(), c2.isRunnable, c1.run(), c2.run()

that would fail fast, however it turns out c2.isRunnable more often than not requires outputs from c1.run() so I had change the flow to

composite.isRunnable(), c1.isRunnable(), c1.run(), c2.isRunnable(), c2.run()

So no you would get deep into the chain before validation would happen.

Reply:
If it is just a single hack, maybe it is fine.

The right way would be to update commands to have two stage validation:
1. initialArgValidation() - can be run on chained commands before running any command
2. preRunValiation() - ran directly before the current command is ran

Or to some extent, JCommander already does 1 above, we could just add a custom validator to the Parameter for the class.

Reply:
Yeah, its a one off validation, but it would be nice to validate the yaml in the create env composite command isRunnable(). Something to think of.